src/containers/ProfilePage/index.jsx

```diff
@@ -0,0 +1,98 @@
+import React from "react";
+import {
+  Container,
+  Row,
+  Accordion,
+  useAccordionToggle,
+  AccordionContext,
+} from "react-bootstrap";
+import { useForm } from "react-hook-form";
+import UserDetailForm from "../../components/profile/UserDetailForm";
+import UserEmailForm from "../../components/profile/UserEmailForm";
+import UserPasswordForm from "../../components/profile/UserPasswordForm";
+import defaultProfileImage from "../../images/defaultProfileImage.png";
+import "./index.scss";
+
+const SECTINOS = [
+  {
+    label: "User details",
+    componentName: UserDetailForm,
+  },
+  {
+    label: "Email address",
+    componentName: UserEmailForm,
+  },
+  {
+    label: "Change password",
+    componentName: UserPasswordForm,
+  },
+];
+
+function getArrowClass(isCurrentElementSelected) {
+  return isCurrentElementSelected ? "arrow down" : "arrow right";
+}
+
+export default function ProfilePage({ submit }) {
+  const { register, handleSubmit, errors } = useForm();
+
+  const onSave = (data) => {
+    // TODO: save data in the database
+    submit(data);
+  };
+
+  function ToggleButton({ children, eventKey, callback }) {
+    const currentEventKey = React.useContext(AccordionContext);
+
+    const decoratedOnClick = useAccordionToggle(
+      eventKey,
+      () => callback && callback(eventKey)
+    );
+
+    const isCurrentEventKey = currentEventKey === eventKey;
+
+    return (
+      <button type="button" onClick={decoratedOnClick} className="sectionTitle">
+        <i className={getArrowClass(isCurrentEventKey)} />
+        {children}
+      </button>
+    );
+  }
+
+  const Section = ({ eventKey, label, Component }) => (
+    <Row className="section">
+      <ToggleButton eventKey={eventKey}>{label}</ToggleButton>
+      <div className="sectionContent">
+        <Accordion.Collapse eventKey={eventKey}>
+          <Component register={register} validErrors={errors} />
```

Comment:
where is register and errors coming from? a component should not be using stuff in scope of another component.

I'd replace this line with {children}

forcing it to take these props is a bit overly specific